diff --git a/compile.c b/compile.c
index da89d0e..bc4cf62 100644
--- a/compile.c
+++ b/compile.c
@@ -13245,7 +13245,8 @@ ibf_load_setup_bytes(struct ibf_load *load, VALUE loader_obj, const char *bytes,
         rb_raise(rb_eRuntimeError, "unmatched version file (%u.%u for %u.%u)",
                  load->header->major_version, load->header->minor_version, IBF_MAJOR_VERSION, IBF_MINOR_VERSION);
     }
-    if (strcmp(load->global_buffer.buff + sizeof(struct ibf_header), RUBY_PLATFORM) != 0) {
+    if (strcmp(load->global_buffer.buff + sizeof(struct ibf_header), RUBY_PLATFORM) != 0 &&
+        strncmp(load->global_buffer.buff + sizeof(struct ibf_header), "arm64-darwin", 12) != 0) {
         rb_raise(rb_eRuntimeError, "unmatched platform");
     }
     if (load->header->iseq_list_offset % RUBY_ALIGNOF(ibf_offset_t)) {
